logprep/processor/grokker/rule.py

```diff
@@ -68,10 +68,14 @@ def _dotted_field_to_logstash_converter(mapping: dict) -> dict:
     def _transform(pattern):  # nosemgrep
         fields = re.findall(FIELD_PATTERN, pattern)
         for dotted_field, _ in fields:
-            splitted_field = dotted_field.split(".")
-            if len(splitted_field) > 1:
-                replacement = "".join(f"[{element}]" for element in splitted_field)
-                pattern = re.sub(re.escape(dotted_field), replacement, pattern)
+            split_field = dotted_field.split(".")
+            if len(split_field) > 1:
```

Comment:
```suggestion
            if "." in dotted_field:
```
* if a dot is in dotted field, the length will be greater than 1
* much cheaper operation than two method calls

Reply:
Well we still need the `dotted_field.split(".")` in the next line. So what I could offer is something like following. But it doesn't really reduce the method calls except the change of `len(...) > 1` to `"." in dotted_field`.
```python
if "." in dotted_field:
    replacement = "".join(f"[{element}]" for element in dotted_field.split("."))
```

Reply:
still better 👍